### PR TITLE
Add ability to enable multitenancy without gateway

### PR DESCRIPTION
**Pull request by rubenvp8510**
Fixes https://github.com/grafana/tempo-operator/issues/224

### CODE REVIEW

**Review by iblancasa — COMMENTED** [review]
Some unit tests are failing.

Also: do we want to add some kind of step to the E2E tests to verify this or do we think it will be enough with the unit tests?

File: internal/manifests/gateway/gateway.go
Comment (andreasgerstmayr) [inline internal/manifests/gateway/gateway.go:40]: Personally I wouldn't call `BuildGateway()` in `BuildAll()` if `params.Tempo.Spec.Template.Gateway.Enabled == false`

**Review by andreasgerstmayr — COMMENTED** [review]
LGTM once the unit tests pass

**Comment by rubenvp8510** [comment]
> Some unit tests are failing.
> 
> Also: do we want to add some kind of step to the E2E tests to verify this or do we think it will be enough with the unit tests?

Not sure, let me think if we can add an e2e test.

**Review by andreasgerstmayr — APPROVED** [review]

**Comment by rubenvp8510** [comment]
Wait a little bit until merge this, I'd like to do a final test :)

**Comment by rubenvp8510** [comment]
This is ready to merge, I've already test it, not sure how to add

**Pull request merged**
